src/generic/MPoly.jl

```diff
@@ -497,6 +497,18 @@ function Base.hash(x::MPoly{T}, h::UInt) where {T <: RingElement}
    return b
 end
 
+# Fallback hash function for multivariate polynomials implemeting the
+# iterators
+function Base.hash(x::MPolyElem{T}, h::UInt) where {T <: RingElement}
+   b = 0x53dd43cd511044d1%UInt
+   for (e, c) in zip(exponent_vectors(x), coeffs(x))
+      b = xor(b, xor(hash(c, h), h))
```

Comment:
`xor` accepts a variable number of arguments, this can be simplified to `b = xor(b, hash(c, h), h)`. BTW, any reason why you include `h` in the `xor`, as it's already consumed in `hash(c, h)` ?

Reply:
I just copy it from the other hash function.